services/base-jdbc/src/main/java/org/eclipse/hono/service/base/jdbc/store/device/TableAdapterStore.java

```diff
@@ -114,4 +195,52 @@ public TableAdapterStore(final SQLClient client, final Tracer tracer, final Stat
 
     }
 
+    /**
+     * Resolve a list of group members.
+     *
+     * @param tenantId The tenant the device belongs to.
+     * @param viaGroups The viaGroups list of a device. This list contains the ids of groups.
+     * @param spanContext The span to contribute to.
+     *
+     * @return A future tracking the outcome of the operation.
+     *
+     * @see org.eclipse.hono.deviceregistry.service.device.AbstractRegistrationService#resolveGroupMembers(String, JsonArray, Span)
+     */
+    public Future<Set<String>> resolveGroupMembers(final String tenantId, final Set<String> viaGroups, final SpanContext spanContext) {
+
+        final Span span = TracingHelper.buildChildSpan(this.tracer, spanContext, "find credentials", getClass().getSimpleName())
+                .withTag("tenant_instance_id", tenantId)
```

Comment:
The tenant ID gets translated by the `tenantExists` method to the `TenantKey`, which may translate the tenant id, into some internal ID. So it may (or may not) be the original tenant id. Thus the addition of "instance".

Reply:
Having a look at the call hierarchy to this method here, the main path is leading down from `AbstractRegistrationService#assertRegistration`, via the `createSuccessfulRegistrationResult` method:
https://github.com/eclipse/hono/blob/27f06c09b9de76b56e30f48cb5c82949d228c005/services/device-registry-base/src/main/java/org/eclipse/hono/deviceregistry/service/device/AbstractRegistrationService.java#L125-L146
In that excerpt it looks like the tenant id that is passed on as `createSuccessfulRegistrationResult` parameter and that ends up as `resolveGroupMembers` parameter above is the _original_ (outward-facing) tenantId from _before_ the `tenantExists` invocation. But seeing this id being used in the SQL statement means it probably should really be the internal id?

In any case, if we have methods inside the data store implementation that deal with an _internal_ tenant id, I think we should make that clear by using another variable name like `tenantInternalId` or something like that. `TenantKey.tenantId` and the getter should then be renamed as well IMHO (the same in `DeviceKey` and `CredentialKey`). Otherwise this will lead to confusion like with the above method.